runtime/compiler/trj9/env/VMJ9.h

```diff
@@ -504,12 +504,13 @@ class TR_J9VMBase : public TR_FrontEnd
 
    uint32_t                   getAllocationSize(TR::StaticSymbol *classSym, TR_OpaqueClassBlock * clazz);
 
-   virtual TR_OpaqueClassBlock *getObjectClass(uintptrj_t objectPointer);
-   virtual uintptrj_t           getReferenceFieldAt(uintptrj_t objectPointer, uintptrj_t offsetFromHeader);
-   virtual uintptrj_t           getVolatileReferenceFieldAt(uintptrj_t objectPointer, uintptrj_t offsetFromHeader);
-   virtual uintptrj_t           getReferenceFieldAtAddress(uintptrj_t fieldAddress);
-   virtual uintptrj_t           getReferenceFieldAtAddress(void *fieldAddress){ return getReferenceFieldAtAddress((uintptrj_t)fieldAddress); }
-   int32_t                      getInt32FieldAt(uintptrj_t objectPointer, uintptrj_t fieldOffset);
+   TR_OpaqueClassBlock *getObjectClass(uintptrj_t objectPointer);
+   uintptrj_t           getReferenceFieldAt(uintptrj_t objectPointer, uintptrj_t offsetFromHeader);
+   uintptrj_t           getVolatileReferenceFieldAt(uintptrj_t objectPointer, uintptrj_t offsetFromHeader);
+   uintptrj_t           getReferenceFieldAtAddress(uintptrj_t fieldAddress);
+   uintptrj_t           getReferenceFieldAtAddress(void *fieldAddress){ return getReferenceFieldAtAddress((uintptrj_t)fieldAddress); }
+   uintptrj_t           getStaticReferenceFieldAtAddress(uintptrj_t fieldAddress, TR_OpaqueClassBlock *clazz = NULL);
```

Comment:
This API worries me - the VM API seems to do nothing with the clazz it is passed and we are omitting sending the class through because it is hard to find/we don't always have it. Do we really need to pass the class through? If it is a future potential use case where we might need it why not keep the API simpler until we do need it? As it is the class clutters the code and is not being checked so is giving the false sense of security that it is being generated and that it is even right. Unused code rots and confuses things so wouldn't it be cleaner to just drop the pretense for now?

Reply:
100% agree. This is the exact same conclusion as in https://github.com/eclipse/openj9/pull/1047#discussion_r171374224. We should try not to speculate future APIs needed to do something. Simplify things now and let this future person introduce the parameter if they actually need to use it. This future use may not ever materialize and we'll end up paying the technical debt for a long time.

Reply:
I'm on a fence. The code lived so long with treating the static reference as a raw pointer, and this is a step forward. On the other side, I do indeed see no obvious (other that some debug/redundancy check) scenario where the class pointer is really relevant. 

@charliegracie, @DanHeidinga  What do you think:
1) about static field access API:
  J9STATIC_OBJECT_LOAD/STORE(thread, clazz, address),
where address is an absolute address of the static.
Is/could class pointer be relevant?
2) if not relevant, how about simplifying it (or providing just another generic API without object/class ?
3) about keeping API and providing NULL, where class pointer is not trivial to provide?